services/gateway/gateway-service.mjs

```diff
@@ -166,6 +166,21 @@ class GatewayService {
     let firstLocation = {playerID: playerID, displayName:
       socket.playerData.displayName};
     firstLocation.mapCoord = socket.playerData.mapCoord;
+    firstLocation.x = socket.playerData.x;
+    firstLocation.y = socket.playerData.y;
+    if(socket.playerData.x == -1 && socket.playerData.y == -1){
+      let set_location = -1, index = 0;
+      let spawn_point = this.gameMap.getSpawnPoint(firstLocation.mapCoord);
+      spawn_point.sort(() => Math.random() - 0.5);
+      while(set_location == -1 && index < spawn_point.length){
+        set_location = 1 ;/*set spawn point */
+        if(set_location != -1){
+          firstLocation.x = spawn_point[index].x;
+          firstLocation.y = spawn_point[index].y;
+        }
+        index = index + 1;
```

Comment:
Error handling if no location is found? (i.e. disconnect the player or just let it wait?)